src/server/server.ml

```diff
@@ -0,0 +1,100 @@
+(*
+  This file is part of scilla.
+
+  Copyright (c) 2018 - present Zilliqa Research Pvt. Ltd.
+
+  scilla is free software: you can redistribute it and/or modify it under the
+  terms of the GNU General Public License as published by the Free Software
+  Foundation, either version 3 of the License, or (at your option) any later
+  version.
+
+  scilla is distributed in the hope that it will be useful, but WITHOUT ANY
+  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+  A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License along with
+  scilla.  If not, see <http://www.gnu.org/licenses/>.
+*)
+
+open Core
+open DebugMessage
+open ErrorUtils
+open Api
+
+(* You can swap the RPC engine, by using a different monad here,
+   note however that if you are using an asynchronous one, like
+   lwt or async, you should also use their specific IO functions
+   including the print functions.
+
+   You can easily put [ExnM] here and the code would stay unchanged. *)
+module M = Idl.IdM
+module IDL = Idl.Make (M)
+
+module Server = API (IDL.GenServer ())
+
+(* Makes a handler that executes the given [callback] with [args],
+   caches the result using the LRU cache and returns it. **)
+let mk_handler callback args =
+  let open IDL.ErrM in
+  try
+    let result = callback @@ Some args in
+    (* TODO: implement AST caching here *)
```

Comment:
I don't think AST caching should be here. Let's remove this comment (and the one mentioning caches at the beginning of this function)